test/run_unit.sh.in

```diff
@@ -112,5 +114,6 @@ else
 	echo "============================================================="
 	echo ":test-result: FAIL $tst" >> $test_output
 fi
+exit 1
```

Comment:
Ah, hm, one side-effect of this is that now we stop on the first failure with 'make check', while we used to run all tests and we only reported an error at the very end with all failures. Is this wanted?

I get that it's not nice to grep for an error string in the log, but possibly it's a smaller problem than only seeing till the next test failure now. What do you think?

Reply:
Agreed. `--fail-fast` was added in 6de46d746e4932b7aa339fb251fc6b5f5cc51413 specifically to address this point.